application/src/main/java/uk/nhs/hee/tis/revalidation/service/RecommendationElasticSearchService.java

```diff
@@ -50,4 +52,83 @@ public void addRecommendationViews(RecommendationView dataToSave) {
           dataToSave.getGmcReferenceNumber(), dataToSave.getTcsPersonId(), ex);
     }
   }
+
+  /**
+   * add new Recommendation trainee to elasticsearch index.
+   *
+   * @param dataToSave Recommendation trainee to go in elasticsearch
+   */
+  public void saveRecommendationViews(RecommendationView dataToSave) {
+    Iterable<RecommendationView> existingRecords = findRecommendationViewsByGmcNumberPersonId(
+        dataToSave);
+
+    // if doctor already exists in ES index, then update the existing record
+    if (Iterables.size(existingRecords) > 0) {
+      updateRecommendationViews(existingRecords, dataToSave);
+    }
+    // otherwise, add a new record
+    else {
+      addRecommendationViews(dataToSave);
+    }
+  }
+
+  /**
+   * update existing Recommendation to elasticsearch index.
+   *
+   * @param existingRecords existing Recommendation to be updated in elasticsearch
+   * @param dataToSave      new Recommendation details to be saved in elasticsearch
+   */
+  private void updateRecommendationViews(Iterable<RecommendationView> existingRecords,
+      RecommendationView dataToSave) {
+    existingRecords.forEach(connectedView -> {
+      dataToSave.setId(connectedView.getId());
+      try {
+        recommendationElasticSearchRepository.save(dataToSave);
+      } catch (Exception ex) {
+        LOG.info("Exception in `updateConnectedViews` (GmcId: {}; PersonId: {}): {}",
+            dataToSave.getGmcReferenceNumber(), dataToSave.getTcsPersonId(), ex);
+      }
+    });
+  }
+
+  /**
+   * find iterable of RecommendationView from elasticsearch index.
+   *
+   * @param dataToSave RecommendationView to go in elasticsearch
+   */
+  private Iterable<RecommendationView> findRecommendationViewsByGmcNumberPersonId(
+      RecommendationView dataToSave) {
```

Comment:
just to make it more clear can you just pass the parameters "gmcReferenceNumber" and "TcsPersonId" - when I first read this I thought this method was also saving stuff 😁